models/consts/consts.go

```diff
@@ -23,10 +23,11 @@ var (
 	Offline string = "offline"
 
 	// proxy type
-	TcpProxy   string = "tcp"
-	UdpProxy   string = "udp"
-	HttpProxy  string = "http"
-	HttpsProxy string = "https"
-	StcpProxy  string = "stcp"
-	XtcpProxy  string = "xtcp"
+	TcpProxy       string = "tcp"
+	UdpProxy       string = "udp"
+	HttpProxy      string = "http"
+	HttpsProxy     string = "https"
+	StcpProxy      string = "stcp"
+	XtcpProxy      string = "xtcp"
+	WebsocketProxy string = "websocket"
```

Comment:
Nothing about websocket proxy, you should only change codes about `protocol` how frpc connect to frps.

Reply:
If reuse the bind_port, the websocket is use http,  when the bind_port == vhost_http_port, how to differentiate the request conn is login or is for vhost_http ?

Reply:
Yes, it's a good question and i will try to do that after this PR.